DEVELOPMENT.md

```diff
@@ -1,119 +1,238 @@
---environment=production`. The CSS generated from the Sass source varies based
-on the environment; for production, it will be compressed, otherwise it is more
-readable and includes line numbers mapping to the original source files. The
-generated HTML also uses the environment to load the appropriate configuration
-from `src/conf/`.
+* [Official documentation for ESLint configuration ](https://eslint.org/docs/user-guide/configuring/)
+* [plugins to run ESLint in various editors and IDEs](https://eslint.org/docs/user-guide/integrations#editors)
 
-## Running the app locally
+# Running the App Locally
 
-Make sure grunt is installed by running `npm install -g grunt-cli` (with root
-access).
+Run a local server instance at `http://localhost:8080`:
 
-Run `grunt build:nsidc-dev` or `grunt build:ade-dev` to generate the
-appropriate `src/index.html` file (as well as css files), then
-`./run_local_webserver.rb` and open your browser to http://localhost:8081.
+        npm start
 
-To run using the code generated by the RequireJS optimizer (concatenated and
-minified js, rather than the standard source), modify
-`local_webserver_config.yaml` to use `build/` instead of `src/`.
+By default, the local server will set the environment to `development`, and will
+use the production OpenSearch endpoints. See the OpenSearch endpoint configuration
+in `src/config/appConfig.js`.
 
-### External services
+# Running on a VM
 
-The external services used by the search portal are defined in
-`local_webserver_config.yaml`. `search_proxies` are the default choice, and
-contain endpoints for the services running on `localhost`. `integration_proxies`
-and `qa_proxies` are also defined.
+## Building application artifacts for deployment to developer VMs
 
-To run against services on integration:
+1. Build the application artifacts into the `dist` directory:
 
-```bash
-./run_local_webserver.rb 8081 integration_proxies
-```
+        npm run build # Build a bundle packaged for production environment
+        npm run build:dev  # Build with source maps for development environment
+                           # Update environment-dependent URL configurations in
+                           # `src/config/appConfig.js` as needed.
 
-To run against services on qa:
+2. To remove old build artifacts from `dist`:
 
-```bash
-./run_local_webserver.rb 8081 qa_proxies
-```
+        npm run clean
+
+## Deploying to a developer VM (without Drupal)
+
+`Puppet` configuration exists in this project to build an application VM, as well as a CI
+machine. All the usual `vagrant-nsidc` environments are supported, but note that a blue
+machine will never be necessary except perhaps for testing purposes. The search interface
+is not deployed via a VM blue-green swap, but rather by pushing a Javascript bundle to
+[npmjs.com](www.npmjs.com).
+
+To create a developer VM:
+
+    vagrant nsidc up --env=dev
+
+TODO: `/opt/nsidc_search` is not being created/populated automatically during VM
```

Comment:
Is there a story to address this?  I didn't see one, but maybe I overlooked it.  It *should* be pretty easy, I think.

Reply:
I don't know! Since the app doesn't run in a VM environment it was a low-priority issue at the time. Developers are impacted but not the production deploy. This branch represents the "good enough" changes to get search running in Drupal 9. It will have to be "good enough" for a while longer.

Reply:
SRCH-136 created.

Reply:
Even if it's a "do sometime down the road" thing, at least it doesn't get lost :)